operator/src/main/java/oracle/kubernetes/operator/helpers/ResponseStep.java

```diff
@@ -151,8 +151,11 @@ public NextAction onFailure(Packet packet, CallResponse<T> callResponse) {
    * @return Next action for fiber processing, which may be a retry
    */
   public NextAction onFailure(Step conflictStep, Packet packet, CallResponse<T> callResponse) {
-    return Optional.ofNullable(doPotentialRetry(conflictStep, packet, callResponse))
-        .orElse(onFailureNoRetry(packet, callResponse));
+    Optional<NextAction> optionalNextAction =
```

Comment:
New functionality without a unit test

Reply:
This is a bug fix and not a new functionality. Previously, "onFailureNoRetry()" method in orElse clause was getting called even when Optional.ofNullable(doPotentialRetry()) returned a non-empty Optional. The change is to make sure that onFailureNoRetry() is called when Optional.ofNullable(doPotentialRetry()) returns empty optional.